### PR TITLE
New function to return sparql query results as list of vim dictionaries

### PR DESCRIPTION
**Pull request by grechaw**
Check out the graphmart query and the function vimanzo#query#queryForVim
I thought this would be a good utility function on which to build navigation trees.

If Anzo supported JSON-LD I would have done construct queries but this is a pretty nice result set too.

### CODE REVIEW

File: autoload/vimanzo/query.vim [hunk 1]
```diff
@@ -26,23 +26,40 @@ function! vimanzo#query#GetGraph(uri)
...
-    silent !clear
-    execute "!" . g:anzo_command . " query -z " . g:anzo_settings . " -f " . bufname("%")
+"This function executes a sparql query and unpacks the json
+"to reutrn the results as a vim structure
+function! vimanzo#query#queryForVim()
+  let l:query_file = bufname("%")
+  " Use json results and all graphs
+  let l:query_options="-a -o json"
+  let l:result_string=system(g:anzo_command . " query " . l:query_options . " -z " . g:anzo_settings . " -f " . l:query_file)
+  let l:result_list=json_decode(l:result_string)['results']['bindings']
```
Comment (andrewppar) [inline autoload/vimanzo/query.vim:166]: I didn't know json_decode was a vim built in function.

File: autoload/vimanzo/query.vim [hunk 2]
```diff
@@ -26,23 +26,40 @@ function! vimanzo#query#GetGraph(uri)
...
 
-function! vimanzo#query#ExecuteQuery()
-    silent !clear
-    execute "!" . g:anzo_command . " query -z " . g:anzo_settings . " -f " . bufname("%")
+"This function executes a sparql query and unpacks the json
+"to reutrn the results as a vim structure
+function! vimanzo#query#queryForVim()
+  let l:query_file = bufname("%")
+  " Use json results and all graphs
+  let l:query_options="-a -o json"
```
Comment (andrewppar) [inline autoload/vimanzo/query.vim:164]: One thing that might stop things up here is that I think this option errors if you have a construct query.

Comment (grechaw) [reply autoload/vimanzo/query.vim:164]: true -- this is only for select and probably just for internal functions.  After I merge your other PR I'll make an issue to refactor the graphmart query to use this.

**Comment by grechaw** [comment]
I just did a force push on this branch to incorporate your changes @andrewppar -- our coding styles have diverged a little but that's to be expected...  love the graphmart selection

**Pull request merged**